### ISSUE TITLE
in_mem reports false memory statistics when a process name contains a parenthesis.

### ISSUE THREAD

**Issue by fujimotos** [fluent/fluent-bit#766]
## Bug Report

**Describe the bug**

`in_mem` reports false memory statistics when the process name contains a parenthesis.

**To Reproduce**

 1. Install fluent-bit on Debian 9.2 
 2. Find out the pid of '(sd-pam)' process

        $ PAM_PID=`pgrep '(sd-pam)'`

 3. Execute the following command:

        $ fluent-bit -i mem -p "pid=$PAM_PID" -o stdout

**Bug**

fluent-bit always reports zero bytes regardless of the real memory usage of the process.

```
[0] mem.0: [1536913471.000270704, {"Mem.total"=>1020300, "Mem.used"=>766768, "Mem.free"=>253532, "Swap.total"=>1046524, "Swap.used"=>4, "Swap.free"=>1046520, "proc_bytes"=>0, "proc_hr"=>"0 b"}]
```

The actual content of `/proc/${PAM_PID}/stat` is here:

```
830 ((sd-pam)) S 829 829 829 0 -1 1077936448 39 0 0 0 0 0 0 0 20 0 1 0 339940 86597632 411 18446744073709551615 1 1 0 0 0 0 0 4096 0 0 0 0 17 1 0 0 0 0 0 0 0 0 0 0 0 0 0
```

**Expected behavior**

`proc_bytes` and `proc_hr` should contain the actual memory usage of the process.

**Your Environment**

* Fluent-bit v0.14.1
* Debian 9.2

**Comment by nokute78** [comment]
Would you copy/paste  result of `pgrep '(sd-pam)'` ? 
And try to use `pgrep '\(sd-pam\)' ?

I tried pgrep on CentOS6.
`pgrep '(aaa)'` did not work correctly.
`pgrep '\(aaa\)'` did work.

**Comment by fujimotos** [comment]
If you suspect that `pgrep` is causing the problem, it's not:

    $ pgrep '(sd-pam)'
    747
    $ ps -p 747
    PID TTY          TIME CMD
    747 ?        00:00:00 (sd-pam)

(Actually I did not use pgrep when I first encountered this bug.
I included pgrep just as a help in making this issue more easily
reproducable for you)

My though is that `in_mem/proc.c` is broken. I haven't studied the
logic throughly yet, but my suspicion is that the logic in `proc_stat()`
really does not handle the case well.
